lightning/src/ln/channelmanager.rs

```diff
@@ -429,6 +439,10 @@ pub struct ChannelManager<ChanSigner: ChannelKeys, M: Deref, T: Deref, K: Deref,
 	/// Taken first everywhere where we are making changes before any other locks.
 	total_consistency_lock: RwLock<()>,
 
+	/// Used to signal to the ChannelManager persister that the manager needs to be re-persisted to
+	/// disk/backups.
+	pub persistence_lock: Arc<(Mutex<bool>, Condvar)>,
```

Comment:
Obviously needs a sample of how to use this in the docs :). I think maybe it would be better to expose a clean `block_until_needs_persist(Option<Max Wait Time>)` function, but there may be missing use cases there, not 100% sure.

Reply:
Added this, I think it made it way nicer